src/continuation/continuation_featurizing_findandmatch.jl

```diff
@@ -0,0 +1,106 @@
+end
+
+"""
+Continuation here is very similar to the one done with recurrences. The difference is only
+in how the ics from previous attractors are seeded to new parameters. In this case, we get ics sampled the previous attractors and pass them to 
+basins_fractions, which extracts features from them and pushes them together with the other features. 
+This could be generalized somehow so that one function could deal with both of the mappers, reducing this code duplication.
+"""
+function continuation(
+        fam::FeaturizingFindAndMatch,
+        prange, pidx, ics;
+        samples_per_parameter = 100, show_progress = true,
+    )
+    progress = ProgressMeter.Progress(length(prange);
+        desc="Continuating basins fractions:", enabled=show_progress
+    )
+
+    if ics isa Function
+        error("`ics` needs to be a Dataset.")
+    end
+
+    (; mapper, distance, threshold) = fam
+    reset!(mapper)
+    # first parameter is run in isolation, as it has no prior to seed from
+    set_parameter!(mapper.ds, pidx, prange[1])
+    fs, _ = basins_fractions(mapper, ics; show_progress = false, N = samples_per_parameter)
+    # At each parmaeter `p`, a dictionary mapping attractor ID to fraction is created.
+    fractions_curves = [fs]
+    # Furthermore some info about the attractors is stored and returned
+    prev_attractors = deepcopy(mapper.attractors)
+    get_info = attractors -> Dict(
+        k => fam.info_extraction(att) for (k, att) in attractors
+    )
+    info = get_info(prev_attractors)
+    attractors_info = [info]
+    ProgressMeter.next!(progress; showvalues = [("previous parameter", prange[1]),])
+    # Continue loop over all remaining parameters
+    for p in prange[2:end]
+        set_parameter!(mapper.ds, pidx, p)
+        reset!(mapper)
+        
+        # Collect ics from previous attractors to pass as additional ics to basins fractions (seeding)
+        # to ensure that the clustering will identify them as clusters, we need to guarantee that there
+        # are at least `min_neighbors` entries
+        additional_ics = Dataset(vcat(map(att-> 
+            fam.seeds_from_attractor(att, fam.mapper.group_config.min_neighbors),
+            values(prev_attractors))...)) #dataset with ics seeded from previous attractors
+        
+        # Now perform basin fractions estimation as normal, utilizing found attractors
+        fs, _ = basins_fractions(mapper, ics;
+            show_progress = false, N = samples_per_parameter, additional_ics
+        )
+        current_attractors = mapper.attractors
+        if !isempty(current_attractors) && !isempty(prev_attractors)
+            # If there are any attractors,
+            # match with previous attractors before storing anything!
+            rmap = match_attractor_ids!(
+                current_attractors, prev_attractors; distance, threshold
```

Comment:
In this method you compare de distance between trajectories that spans the attractors. Can we compute the distance between each clusters of points in the feature space? Is it more reliable than matching trajectories?

Reply:
So matching by comparing distances between the features, instead of attractors? It's a nice idea. As I understand, this could be done by the user via the `distance` function, right? Receive the attractor, extract the feature and compute the distance. 

One advantage of matching directly by the features is that one wouldn't need to reconstruct the attractors, and so could still use `ics` as a sampler function. But I think the generality of having the attractors (generated by having ìcs` as a dataset) outweights the memory savings of having `ics` as a function. What do you think?

Reply:
> So matching by comparing distances between the features, instead of attractors? It's a nice idea. As 

@KalelR that's what I was trying to tell you in-person in Portland... That you convert the clusters into `StateSpaceSet` and then call the `match_attractor_ids!` function. The function doesn't know what an attractor is; it could be an actual attractor or any `StateSpaceSet`.

Reply:
With this matching process, you completely alleviate the need to re-construct the attractors and to store/find an initial condition to go to each attravtcort. Yet you still keep the slice-by-slice continuation in featurizing that you want.

Reply:
You can match a subset of the cluster to avoid a memory bottleneck.

Reply:
> you convert the clusters into StateSpaceSet and then call the match_attractor_ids! function on the clusters themselves 

Then the comparison is always done on the features, correct? My point is that, if you reconstruct the attractors, you can still do that (just extract the features again) but also much more (simple example being Euclidean distance). 

My question then is: do you guys think that this is not worth it? That if the attractors have been identified by clusterizing the features, then they also should be matched by comparing the features? 

> You can match a subset of the cluster to avoid a memory bottleneck.
Do you mean e.g. only taking one feature to represent the cluster (i.e. attractor) for the matching process?

Reply:
> point is that, if you reconstruct the attractors, you can still do that (just extract the features again) but also much more (simple example being Euclidean distance). 

Yes, I understand this. What I am not sure is whether there is benefit in mixing the two approaches in this way. The benefit of the recurrence approach is the rigorousness that you have a guarantee that you find an attractor, while for featurizing cluster != attractor.

> do you guys think that this is not worth it? That if the attractors have been identified by clusterizing the features, then they also should be matched by comparing the features?

I am suggesting that, however I would like first to see results of matching directly on the feature space before converting to attractors. I think this is the best option. But I could be wrong, and what you prpose, to go for feature space to attractor space, then match, and then go back again, could be a better approach. It is much more complicated though both logically and in terms of code. So I would still argue with should first try parameter-by-parameter matching directly on the feature space, simply by considering each cluster / group-of-features as a StateSpaceSet and then literally calling the `match_attractor_ids!` function.

Reply:
p.s.: Sorry for the late reply.

Reply:
Is there a metric to measure how good is the matching between two clusters of features? For example [distance correlation](https://en.wikipedia.org/wiki/Distance_correlation) would be an example.

Reply:
good idea!

Reply:
Actually, I'm not sure because this measure is invariant w.r.t. absolute position of the clusters. While for us this is crucial. If one cluster is identical to the other but shifted by 10 in each dimension, these should be totally different clusters / attractors.